src/excel/buildExcel.ts

```diff
@@ -0,0 +1,136 @@
+import { WorkBook } from "xlsx";
+import * as XLSX from "xlsx";
+import { Dict } from "../types/utilTypes";
+import { ColorType, FeatureIndicators, MapFeature } from "../types/resourceTypes";
+import { useIndicatorColors } from "../composables/useIndicatorColors";
+
+export class BuildExcel {
+    private readonly _appConfig: AppConfig;
+
+    private readonly _countryNames: Dict<string>;
+
+    private readonly _admin1Indicators: Dict<FeatureIndicators>;
+
+    private readonly _admin2Indicators: Dict<FeatureIndicators>;
+
+    private readonly _admin1Geojson: Dict<MapFeature[]>;
+
+    private readonly _admin2Geojson: Dict<MapFeature[]>;
+
+    private readonly _indicatorColors;
+
+    constructor(
+        appConfig: AppConfig,
+        countryNames: Dict<string>,
+        admin1Indicators: Dict<FeatureIndicators>,
+        admin2Indicators: Dict<FeatureIndicators>,
+        admin1Geojson: Dict<MapFeature[]>,
+        admin2Geojson: Dict<MapFeature[]>
+    ) {
+        this._appConfig = appConfig;
+        this._countryNames = countryNames;
+        this._admin1Indicators = admin1Indicators;
+        this._admin2Indicators = admin2Indicators;
+        this._admin1Geojson = admin1Geojson;
+        this._admin2Geojson = admin2Geojson;
+        this._indicatorColors = useIndicatorColors(ref(appConfig));
```

Comment:
or something like that so it matches the function signature
```suggestion
        this._indicatorColors = useIndicatorColors(ref(appConfig), {} as any);
```
maybe if you dont want to use any
```suggestion
        this._indicatorColors = useIndicatorColors(ref(appConfig), computed(() => ({})));
```

Reply:
i have realised we do this in other places too, not too concerned about it haha but yh completely optional